### PR TITLE
Add toDict example to TableWorkspace concept page

### PR DESCRIPTION
**Pull request by samueljackson92**
Add an example to the TableWorkspace concepts page. This should have been part of #21558.

*No associated issue*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
I thought you were going to also add examples for converting to a `pandas.DataFrame`. From the other PR
```python
import pandas as pd
df = pd.DataFrame(table.toDict())
```
It should be marked up as a code block so we don't have to install pandas on the build servers, but people can still see the example. Also, an example that assembles a `pandas.DataFrame` from looping over the columns would be helpful. For example (this likely won't work as is)
```python
df = pandas.DataFrame()
for col in tableWS.columnNames():
    df[col] = tableWS[col]
```

**Comment by samueljackson92** [comment]
@peterfpeterson Sure. I think I misunderstood. I'll add that as well.

File: docs/source/concepts/TableWorkspaces.rst [hunk 1]
```diff
@@ -128,6 +133,26 @@ Table Workspace Properties
...
+.. code-block:: python
+
+    import pandas as pd
+    df = pd.DataFrame(table.toDict())
+
+If only a subset of the data from the table is required, or you're working with an existing :class:`~pandas.DataFrame` and want to append columns from the Table workspace this can be achieved as follows.
+
+.. code-block:: python
+
+    df = pd.DataFrame()
```
Comment (peterfpeterson) [inline docs/source/concepts/TableWorkspaces.rst]: This should be
```python
df = pd.DataFrame()
for col in tableWS.getColumnNames():
    df[col] = tableWS.column(col)
```

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
I know it isn't part of this PR, but please fix up the formatting of the C++ section of the document as well. L172, L176-179, L223-231, and L235-239 should be `code-block:: c` to make the document look nicer.

File: docs/source/concepts/TableWorkspaces.rst [hunk 2]
```diff
@@ -128,6 +133,26 @@ Table Workspace Properties
...
+    for col in tableWS.getColumnNames():
+        df[col] = tableWS[col]
```
Comment (peterfpeterson) [inline docs/source/concepts/TableWorkspaces.rst]: Still a bug. This line should be `df[col] = tableWS.column(col)`

**Review by peterfpeterson — CHANGES_REQUESTED** [review]

**Comment by peterfpeterson** [comment]
Other than the one error in the example, everything else is ready. Good additions.

**Review by peterfpeterson — APPROVED** [review]

**Pull request merged**